pydantic/_internal/_model_construction.py

```diff
@@ -217,7 +219,12 @@ def build_inner_schema(  # noqa: C901
         try:
             default = getattr(cls, ann_name)
         except AttributeError:
-            fields[ann_name] = FieldInfo.from_annotation(ann_type)
+            # if field has no default value and is not in __annotations__ this means that it is
```

Comment:
```suggestion
            # if field has no default value and is not in `type_hints` this means that it is
```

Reply:
Not sure about this change, because `get_type_hints` returns the annotations of inherited members too. I could write `annotations` without dunders, but perhaps i also missunderstand something here...

Reply:
okay, ignore.